metadata-ingestion/src/datahub/sdk/mlmodel.py

```diff
@@ -0,0 +1,293 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Sequence, Type, Union
+
+from typing_extensions import Self
+
+from datahub.emitter.mce_builder import DEFAULT_ENV
+from datahub.metadata.schema_classes import (
+    AspectBag,
+    MLHyperParamClass,
+    MLMetricClass,
+    MLModelPropertiesClass,
+)
+from datahub.metadata.urns import (
+    DataProcessInstanceUrn,
+    MlModelGroupUrn,
+    MlModelUrn,
+    Urn,
+)
+from datahub.sdk._shared import (
+    DomainInputType,
+    HasDomain,
+    HasInstitutionalMemory,
+    HasOwnership,
+    HasPlatformInstance,
+    HasTags,
+    HasTerms,
+    HasVersion,
+    HyperParamsInputType,
+    LinksInputType,
+    OwnersInputType,
+    TagsInputType,
+    TermsInputType,
+    TrainingMetricsInputType,
+    convert_hyper_params,
+    convert_training_metrics,
+    make_time_stamp,
+    parse_time_stamp,
+)
+from datahub.sdk.entity import Entity, ExtraAspectsType
+
+
+class MLModel(
+    HasPlatformInstance,
+    HasOwnership,
+    HasInstitutionalMemory,
+    HasTags,
+    HasTerms,
+    HasDomain,
+    HasVersion,
+    Entity,
+):
+    __slots__ = ()
+
+    @classmethod
+    def get_urn_type(cls) -> Type[MlModelUrn]:
+        return MlModelUrn
+
+    def __init__(
+        self,
+        id: str,
+        platform: str,
+        version: Optional[str] = None,
+        aliases: Optional[List[str]] = None,
+        platform_instance: Optional[str] = None,
+        env: str = DEFAULT_ENV,
+        name: Optional[str] = None,
+        description: Optional[str] = None,
+        training_metrics: Optional[TrainingMetricsInputType] = None,
+        hyper_params: Optional[HyperParamsInputType] = None,
+        external_url: Optional[str] = None,
+        custom_properties: Optional[Dict[str, str]] = None,
+        created: Optional[datetime] = None,
+        last_modified: Optional[datetime] = None,
+        owners: Optional[OwnersInputType] = None,
+        links: Optional[LinksInputType] = None,
+        tags: Optional[TagsInputType] = None,
+        terms: Optional[TermsInputType] = None,
+        domain: Optional[DomainInputType] = None,
+        group: Optional[Union[str, MlModelGroupUrn]] = None,
```

Comment:
maybe this makes sense to change

```suggestion
        model_group: Optional[Union[str, MlModelGroupUrn]] = None,
```